pkg/install/install.go

```diff
@@ -397,9 +397,27 @@ func (i *Installer) deployARMTemplate(ctx context.Context, rg string, tName stri
 		return err == nil, err
 	}, timeoutCtx.Done())
 
-	if isQuota, errMsg := hasResourceQuotaExceededError(err); isQuota {
-		err = api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeQuotaExceeded, errMsg, "")
+	if err != nil {
+		marshaled, marshalErr := json.Marshal(err)
+
+		if marshalErr != nil {
+			i.log.Print(marshalErr)
+			return api.NewCloudError(http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", "Internal server error.")
+		}
+
+		return &api.CloudError{
+			StatusCode: http.StatusBadRequest,
+			CloudErrorBody: &api.CloudErrorBody{
+				Code:    api.CloudErrorCodeDeploymentFailed,
+				Message: fmt.Sprintf("deployment of %s ARM template failed", tName),
```

Comment:
```suggestion
				Message: "ARM template deployment failed.",
```
(user facing error messages start with a capital letter and end in '.'.)